src/main/java/edu/wpi/first/shuffleboard/WidgetPaneController.java

```diff
@@ -257,4 +316,88 @@ private Menu createChangeMenus(WidgetTile tile) {
     return changeView;
   }
 
+  private void moveTile(WidgetTile tile, int count, boolean left) {
+    for (int i = 0; i < count; i++) {
+      Optional<Runnable> move = left ? moveTileLeft(tile) : moveTileUp(tile);
+      if (move.isPresent()) {
+        move.get().run();
+      } else {
+        // Can't move any further, bail
+        break;
+      }
+    }
+  }
+
+  private Optional<Runnable> moveTileLeft(WidgetTile tile) {
+    TileLayout layout = pane.getTileLayout(tile);
+    final int previousColumn = layout.origin.col - 1;
+    if (!pane.isOverlapping(previousColumn, layout.origin.row,
+        layout.size.getWidth(), layout.size.getHeight(), n -> tile == n) && previousColumn >= 0) {
+      // Great, we can move it
+      return Optional.of(() -> GridPane.setColumnIndex(tile, previousColumn));
+    } else if (layout.origin.col > 0) {
+      // Check to see if we can move or shrink any tiles to the left, if possible
+      List<Optional<Runnable>> runs = IntStream.range(layout.origin.row, layout.origin.row + layout.size.getHeight())
+          .mapToObj(row -> pane.tileAt(previousColumn, row))
+          .filter(Optional::isPresent) // guaranteed to be at least one tile
+          .map(Optional::get)
+          .distinct() // need to make sure we have no repeats, or n-row tiles will get moved n times
+          .filter(t -> tile != t)
+          .map(this::moveTileLeft) // recursion here
+          .collect(Collectors.toList());
+      if (runs.stream().allMatch(Optional::isPresent)) {
+        return Optional.of(() -> {
+          runs.forEach(o -> o.get().run());
+          GridPane.setColumnIndex(tile, previousColumn);
+        });
+      } else {
```

Comment:
This still has the shrinking logic in the wrong order. It shrinks things from left-to-right, which is very confusing—you're moving the window from right-to-left. It should shrink the most immediate thing first, and *then* try and recurse.